### PR TITLE
07894 Reconnect Teacher - Throttle Logs and New Metric

### LINKED ISSUE #7894: Throttle teacher's rejected reconnect log messages
When a node attempts to reconnect, if the teacher rejects the request then the teacher writes a message to its log. For example:

```
2023-07-27 14:18:23.553 6938     INFO  RECONNECT        <<platform-core: SyncProtocolWith3 1 to 3>> ReconnectThrottle: Rejecting reconnect request from node 3 due to a previous reconnect attempt at 2023-07-27T19:17:42.586320Z
```

We should attach a rate limiter to this log message to prevent log spam. Perhaps it would even make sense to have one rate limiter per node in the address book, so that we don't lose a log message from a healthy node due to the actions of a thrashing node.

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:

Fixes #7894

### CODE REVIEW

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java [hunk 1]
```diff
@@ -54,12 +68,27 @@ public class ReconnectMetrics {
...
         senderEndTimes = metrics.getOrCreate(SENDER_END_TIMES_CONFIG);
         receiverEndTimes = metrics.getOrCreate(RECEIVER_END_TIMES_CONFIG);
+
+        for (final Address address : addressBook) {
+            final NodeId nodeId = address.getNodeId();
+            rejectionFrequency.put(
+                    nodeId,
+                    new CountPerSecond(
+                            metrics,
+                            new CountPerSecond.Config(
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java:85]: I believe this metric should be given a unit, though beware that the string `\s` is forbidden in unit names, due to prometheus quirks

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/ReconnectProtocol.java [hunk 2]
```diff
@@ -75,7 +84,8 @@ public ReconnectProtocol(
             @NonNull final ReconnectController reconnectController,
             @NonNull final SignedStateValidator validator,
             @NonNull final FallenBehindManager fallenBehindManager,
-            @NonNull final Configuration configuration) {
+            @NonNull final Configuration configuration,
+            @NonNull final Time time) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/ReconnectProtocol.java:88]: missing doc for new param

**Review by alittley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java:85]: adding `withUnits("rejectionsPerSec")`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/ReconnectProtocol.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/ReconnectProtocol.java:88]: added

**Review by alittley — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

**Review by lpetrovic05 — APPROVED** [review]

**Pull request merged**